src/main/java/io/github/ust/mico/kafkafaasconnector/MessageListener.java

```diff
@@ -250,4 +249,15 @@ private ErrorReportMessage getErrorReportMessage(String message, String original
             .setFaasGateway(openFaaSConfig.getGateway())
             .setReportingComponentName("TODO"); //TODO set this via a env variable and use it here
     }
+
+    /**
+     * This method checks if it is necessary to filter it out. This only works
+     * for testMessages. It returns {@code true} if {@code isTestMessage} is
+     * true and the param {@code topic} equals {@code filterOutBeforeTopic}.
+     * @param topic The target topic or destination of this message.
+     * @return isTestMessage && topic.equals(filterOutBeforeTopic)
+     */
+    public boolean isTestMessageCompleted(MicoCloudEventImpl<JsonNode> cloudEvent, String topic) {
+        return cloudEvent.isTestMessage().isPresent() && cloudEvent.isTestMessage().get() && cloudEvent.getFilterOutBeforeTopic().isPresent() && topic.equals(cloudEvent.getFilterOutBeforeTopic().get());
```

Comment:
use orElse(false) and orElse("") for smaller conditions (maybe also check that the given topic is not null or "")